base/dict.jl

```diff
@@ -189,6 +189,13 @@ function grow_to!{K,V}(dest::Associative{K,V}, itr, st)
     return dest
 end
 
+
+"""
+    similar(dictionary)
```

Comment:
In https://github.com/JuliaLang/julia/blob/master/base/abstractarray.jl#L409, `array` is not typed. Would you like me to replace `dictionary` with `d:Dict` nevertheless?